dlrs/main/classes/RollupSummariesSelectorTest.cls

```diff
@@ -109,58 +109,63 @@ private with sharing class RollupSummariesSelectorTest {
 
   @isTest
   private static void selectActiveByUniqueName_notForEdit_MDT() {
-    MockedCustomMetadataSelector mock = new MockedCustomMetadataSelector(false);
-    RollupSummariesSelector.CustomMetadataSelector metadataSelector = mock;
-
-    RollupSummariesSelector rollupSelector = new RollupSummariesSelector(false, false);
-    rollupSelector.m_customMetadataSelector = mock;
+    RollupSummariesSelector rollupSelector = new RollupSummariesSelector(
+      false,
+      false
+    );
+    Assert.areEqual(
+      1,
+      Limits.getQueries(),
+      'Expected a single query against the custom object'
+    );
 
     Test.startTest();
 
     List<RollupSummary> summaries = rollupSelector.selectActiveByUniqueName(
       new Set<String>{ 'A_Summary' }
     );
 
+    Assert.areEqual(0, Limits.getQueries(), 'Expected no additional queries');
     Test.stopTest();
-
-    //No query rows exist in org and query invocations not called. Just check that correct method is used
-    System.assert(
-      mock.getLookupRollupSummaryByDeveloperNameOrIdCalled,
-      'no SOQL queries should have been used'
-    );
   }
 
   @isTest
   private static void selectActiveByUniqueName_ForEdit_MDT() {
-    MockedCustomMetadataSelector mock = new MockedCustomMetadataSelector(false, true);
-    RollupSummariesSelector.CustomMetadataSelector metadataSelector = mock;
-
-    RollupSummariesSelector rollupSelector = new RollupSummariesSelector(false, true);
-    rollupSelector.m_customMetadataSelector = mock;
+    RollupSummariesSelector rollupSelector = new RollupSummariesSelector(
+      false,
+      true
+    );
+    Assert.areEqual(2, Limits.getQueries());
 
     Test.startTest();
 
     //There are no records in the database, just use an empty Set to check against query limits
     List<RollupSummary> summaries = rollupSelector.selectActiveByUniqueName(
       new Set<String>()
     );
-
+    System.assertEquals(0, Limits.getQueries());
     Test.stopTest();
-
-    //No query rows exist in org and query invocations not called. Just check that correct method is used
-    System.assert(
-      mock.getLookupRollupSummariesForEditCalled,
-      'SOQL queries should have been used'
-    );
   }
 
+  // TODO: may not be necessary anymore if it is duplicating work
+  // largely because we don't have a method to inject a mock prior to construction
+  // and now the constructor is building the full object cache and not querying again
   @isTest
   private static void selectActiveByUniqueName_NotForEdit_SortOrder() {
     MockedCustomMetadataSelector mock = new MockedCustomMetadataSelector(false);
     RollupSummariesSelector.CustomMetadataSelector metadataSelector = mock;
 
-    RollupSummariesSelector rollupSelector = new RollupSummariesSelector(false, false);
-    rollupSelector.m_customMetadataSelector = mock;
+    // fill cache
+    RollupSummariesSelector.setRollupCache(
```

Comment:
New test-only means of settings the cache without having to know how the cache key is constructed.